tests/spark_autologging/ml/test_pyspark_ml_autologging.py

```diff
@@ -290,45 +327,150 @@ def test_pipeline(dataset_text):
     inner_pipeline = Pipeline(stages=[hashingTF, lr])
     nested_pipeline = Pipeline(stages=[tokenizer, inner_pipeline])
 
-    assert _get_pipeline_stage_hierarchy(pipeline) == {
-        pipeline.uid: [tokenizer.uid, hashingTF.uid, lr.uid]
-    }
-    assert _get_pipeline_stage_hierarchy(nested_pipeline) == {
-        nested_pipeline.uid: [tokenizer.uid, {inner_pipeline.uid: [hashingTF.uid, lr.uid]}]
-    }
-
     for estimator in [pipeline, nested_pipeline]:
         with mlflow.start_run() as run:
             model = estimator.fit(dataset_text)
+            estimator_info = load_json_artifact("estimator_info.json")
+            metadata = _gen_estimator_metadata(estimator)
+            assert metadata.hierarchy == estimator_info["hierarchy"]
 
+        uid_to_indexed_name_map = metadata.uid_to_indexed_name_map
         run_id = run.info.run_id
         run_data = get_run_data(run_id)
         assert run_data.params == truncate_param_dict(
-            stringify_dict_values(_get_instance_param_map(estimator))
+            stringify_dict_values(_get_instance_param_map(estimator, uid_to_indexed_name_map))
         )
         assert run_data.tags == get_expected_class_tags(estimator)
         assert MODEL_DIR in run_data.artifacts
         loaded_model = load_model_by_run_id(run_id)
         assert loaded_model.uid == model.uid
-        assert run_data.artifacts == ["model", "pipeline_hierarchy.json"]
+        assert run_data.artifacts == ["estimator_info.json", "model"]
+
+
+def test_param_search_estimator(spark_session, dataset_regression):
+    mlflow.pyspark.ml.autolog()
+    lr = LinearRegression(solver="l-bfgs", regParam=0.01)
+    lrParamMaps = [
+        {lr.maxIter: 1, lr.standardization: False},
+        {lr.maxIter: 200, lr.standardization: True},
+        {lr.maxIter: 2, lr.standardization: False},
+    ]
+    best_params = {"LinearRegression.maxIter": 200, "LinearRegression.standardization": True}
+    for ParamSearchEstimator in [CrossValidator, TrainValidationSplit]:
+        # Test on rmse (smaller is better) and r2 (larger is better)
+        for metricName in ["rmse", "r2"]:
```

Comment:
Can we use `pytest.mark.parametrize` here?

```python
@pytest.mark.parametrize("metric_name", [...])
@pytest.mark.parametrize("param_search_estimator", [...])
def test_param_search_estimator(spark_session, dataset_regression, metric_name, param_search_estimator):
```

Reply:
Good idea